cli.js

```diff
@@ -170,11 +166,73 @@ Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
   .command('list [options]', 'List all deployed functions', (yargs0) => {
     yargs0
       .usage('Usage: $0 list [options]')
-      .option('json', { describe: 'Output as JSON' })
+      .option('json', {
+        describe: 'Output as JSON',
+        type: 'boolean',
+      })
       .strict();
   }, async (argv) => {
     await handleCommand(argv, listHandler);
   })
+  .command('triggers', 'Manages triggers', yargs0 => yargs0
+      .demandCommand()
+      .command('upsert', 'Creates a new trigger or updates an existing one', (yargs1) => {
+        yargs1
+          .usage('Usage: $0 triggers upsert <trigger name> [options]')
+          .positional('trigger name', {
+            describe: 'Trigger name',
+            type: 'string',
+          })
+          .option('source-uri', {
+            demandOption: true,
+            describe: 'Redis-stream uri',
+            type: 'string',
+          })
+          .option('target-func', {
+            demandOption: true,
+            describe: 'Name of the function to run for each item in the stream',
+            type: 'string',
+          })
+          .option('error-handler-func', {
+            describe: 'Name of the function to run when an error occurres',
+            type: 'string',
+          })
+          .demandCommand(1)
+          .strict()
+          .example(
+`  // Setup a trigger called 'my-trigger' that will invoke 'my-func' for every entry in 'my-stream'
+  bn triggers upsert my-trigger --source-uri "redis://:redis-password@my-redis.my-domain.com/0#my-stream" --target-func my-func
+
+  // Same this but secured
+  bn triggers upsert my-trigger --source-uri "redis+tls://:redis-password@my-redis.my-domain.com/0#my-stream" --target-func my-func`,
+          );
+      }, async (argv) => {
+        await handleCommand(argv, triggerUpsertHandler);
+      })
+      .command('remove', 'Removes an existing trigger', (yargs1) => {
+        yargs1
+          .usage('Usage: $0 triggers remove <trigger> [options]')
+          .positional('trigger', {
+            describe: 'Trigger name',
+            type: 'string',
+          })
+          .demandCommand(1)
+          .strict();
+      }, async (argv) => {
+        await handleCommand(argv, triggerRemoveHandler);
+      })
+      .command('list', 'Lists all existing triggers', (yargs1) => {
```

Comment:
are you suggesting there will be this commands:
```
bn list // lists functions
bn list --triggers // lists triggers
```
?

(I don't like it...)

Reply:
I'm just saying that right now you don't type 

`bn functions --list`  you type `bn list`

But for triggers you type `bn triggers --list`

Maybe we decide that because functions are a first class citizen they should have a different UX. But as of now, the interface might seem inconsistent to a new user.

FTR: We could always make the default of `bn list` "functions" so `bn list` would still print functions but `bn list --triggers` would print triggers.

Reply:
there is no `bn triggers --list`
it's `bn trigger list`